### PR TITLE
Added proxy commands to valet linux

### PR DESCRIPTION
**Pull request by willishq**
I would have added tests / fixed the test suite but there's more wrong with the tests than can be reasonably fixed in a PR to add proxy support.

### CODE REVIEW

**Comment by willishq** [comment]
#330

File: cli/Valet/CommandLine.php [hunk 1]
```diff
@@ -75,10 +75,6 @@ protected function runCommand($command, callable $onError = null)
         $onError = $onError ?: function () {
         };
 
-        // Symfony's 4.x Process component has deprecated passing a command string
```
Comment (jmarcher) [inline cli/Valet/CommandLine.php:78]: Why did you deleted this comment?

File: cli/Valet/Configuration.php [hunk 2]
```diff
@@ -53,6 +51,15 @@ public function uninstall()
      */
     public function createConfigurationDirectory()
     {
+        $this->files->ensureDirExists(preg_replace('~/valet$~', '', VALET_HOME_PATH), user());
```
Comment (jmarcher) [inline cli/Valet/Configuration.php:54]: Can you explain what this code does or what is it for with comments on your code?

File: cli/Valet/Configuration.php [hunk 3]
```diff
@@ -63,15 +70,15 @@ public function createConfigurationDirectory()
      */
     public function createDriversDirectory()
     {
-        if ($this->files->isDir($driversDirectory = VALET_HOME_PATH . '/Drivers')) {
+        if ($this->files->isDir($driversDirectory = VALET_HOME_PATH.'/Drivers')) {
```
Comment (jmarcher) [inline cli/Valet/Configuration.php:73]: Leave spaces

File: cli/Valet/Configuration.php [hunk 4]
```diff
@@ -63,15 +70,15 @@ public function createConfigurationDirectory()
...
             return;
         }
 
         $this->files->mkdirAsUser($driversDirectory);
 
         $this->files->putAsUser(
-            $driversDirectory . '/SampleValetDriver.php',
-            $this->files->get(__DIR__ . '/../stubs/SampleValetDriver.php')
+            $driversDirectory.'/SampleValetDriver.php',
```
Comment (jmarcher) [inline cli/Valet/Configuration.php:80]: Leave spaces

**Review by jmarcher — COMMENTED** [review]
I do not have time to review the changes right now but as it seems you made a lot of changes but I do not have an idea of what those changes are needed for. Please add some comments.

**Comment by willishq** [comment]
> I do not have time to review the changes right now but as it seems you made a lot of changes but I do not have an idea of what those changes are needed for. Please add some comments.

All I did was copy the functionality from Valet to enable the proxy command to work, adding comments to the changes would basically be "added this method and followed the stack trace, adding methods from Valet as and when required"

Which is basically what I did.

I would suggest that, since this is an actively used repository, you would consider putting out a call for maintainers if you do not have time to maintain it yourself, as there are a lot of people out there who are using this who will not be able to use some features which are now in valet which people would expect to be available.

**Comment by PH7-Jack** [comment]
This pull request is very useful. Thanks @willishq

**Pull request closed without merging**